stonesoup/updater/kalman.py

```diff
@@ -288,6 +332,75 @@ def update(self, hypothesis, **kwargs):
         return self._update_class(posterior_mean, posterior_covariance, hypothesis,
                                   timestamp=hypothesis.measurement.timestamp)
 
+    def soft_update(self, hypothesis, soft_measurement_prediction, force_symmetric_covariance=False, **kwargs):
```

Comment:
I don't think that `soft_measurement_prediction` is needed here. Normally, the measurement prediction is either provided through `hypothesis.measurement_prediction` or, if `hypothesis.measurement_prediction is None` (as per line 366 below), is computer internally. Also see my comment below about the computation of the measurement prediction.

Reply:
Agree. Your comment is applied. To do so, I fixed a mistake in line 89 of hypothesiser\distance.py.